lib/TableUtils.lua

```diff
@@ -36,15 +36,15 @@ function TableUtils.slice(tbl, first, last, step) --: (any[], number?, number?,
 	return sliced
 end
 
-function TableUtils.map(source, handler) --: ((any[], (element: any, key: number) => any) => any[]) | ((table, (element: any, key: string) => any) => table)
+function TableUtils.map(source, handler) --: <T extends Iterable<K,V>, R extends Iterable<U,V>((T, (element: V, key: K) => U) => R)
```

Comment:
Is the type of this correct? Doesn't it just return a `U[]`?

Reply:
No map returns a table if you input a table